### ISSUE TITLE
database is dropped when viewing settings screen if using SD Card sync

### ISSUE THREAD

**Issue by obilodeau** [tomboy-notes/tomdroid#44]
Because when the preference activity loads, setServer is run for initialization and within setServer:

if (!service.needsAuth()){
    resetLocalDatabase();
}

SD Card service doesn't need auth so the database is nuked.

This is all new code to me so I'll see what I can do..

---

Imported from Launchpad using lp2gh.
- date created: 2010-10-09T03:40:25Z
- owner: plaxx
- assignee: plaxx
- the launchpad url was https://bugs.launchpad.net/bugs/657178

**Comment by obilodeau** [comment]
(by obilodeau)
I simply removed the lines. For SD Card it's ok and as far as I can tell it won't do any problems to Web sync... correct?

I'll make my mind on this tomorrow. It's in lp:tomdroid rev 197 for now.

I have an apk signed with debug keys if anyone's willing to test.

**Comment by obilodeau** [comment]
(by trappe)
I've wrote these lines to gain a clean state when switching from Web sync to SD Card. Otherwise your Web notes will still be in the local database.

Unfortunately I overlooked that setServer(...) is not only called when the pref changes but also every time the Activity get's created:

// Enable or disable the server field depending on the selected sync service
        setServer(syncService.getValue());

```
    syncService.setOnPreferenceChangeListener(new OnPreferenceChangeListener() {

        public boolean onPreferenceChange(Preference preference, Object newValue) {

            setServer((String)newValue);
            return true;
        }
    });
```

**Comment by obilodeau** [comment]
(by trappe)
I think there are two simple solutions:
1. move the lines

if (!service.needsAuth()){
    resetLocalDatabase();
}

into the onPreferenceChangeListener
1. find an other way to enable or disable the server field depending on the selected sync service, without calling setServer() in OnCreate.

**Comment by obilodeau** [comment]
(by obilodeau)
Implemented solution 1. onPreferenceChangeListener calls a new syncServiceChanged if service changed then in here I setServer and drop database if !service.needsAuth.

**Comment by obilodeau** [comment]
(by obilodeau)
fix released in 0.4.0
